packages/inferno/src/DOM/utils/common.ts

```diff
@@ -80,12 +90,33 @@ export function findDOMfromVNode(vNode: VNode, startEdge: boolean) {
   return null;
 }
 
-export function removeVNodeDOM(vNode: VNode, parentDOM: Element) {
+export function callAllAnimationHooks(animationQueue: Function[], callback?: Function) {
+  let animsLeft = animationQueue.length;
+  do {
+    const fn = animationQueue.pop();
+    // This check shouldn't be needed but TS complains so adding it
+    if (fn !== undefined) {
+      fn(() => {
+        // When all animations are done, remove everything.
+        // INVESTIGATE: If we add a sibling when the animation is active, will it be removed?
+        if (--animsLeft <= 0 && isFunction(callback)) {
+          callback();
+        }
+      });
+    }
+  } while (animationQueue.length > 0);
+}
+
+function clearVNodeDOM(vNode: VNode, parentDOM: Element, deferedRemoval = false) {
   do {
     const flags = vNode.flags;
 
     if (flags & VNodeFlags.DOMRef) {
-      removeChild(parentDOM, vNode.dom as Element);
+      // On defered removals the node might disappear because of later
+      // operations
```

Comment:
What does it mean: because of later operations? Did you find the node gets de-attached from the DOM?

Reply:
Yes, what happens is that the last item is removed using parent.innerText = '' via clearDOM.

Because transition events can fire out of order in the browser it is unpredictable when this happens during a set of simultaneous animations (that explains the random fail). In Mocha-tests I had to provoke this by having different timeouts because setTimeout fires in a predictable order.

So on deferred removal I need to check that the element still exists. But I don't want to do that check on normal removal since it is a parentNode.contains() check that will incur a performance penalty. Added the deferredRemoval bool to indicate if this check needs to be done so it won't affect performance during synchronous removal.

Reply:
And yes it could happen due to later operations too.

Reply:
hmm, this routine is only called when removing direct children of given parentDOM. 

maybe we could get rid off this query selector by checking 

```
vNode.dom.parentNode === parentDOM 
```

Reply:
or maybe its even enough to just check if 
```
vnode.dom.parentNode !== null
```

Reply:
Smart, yeah I think that would be much better. There is a risk that the dom is moved to descendant child. This was a bad choice from me that could fail in edge cases.

I still believe that we should retain the !deferedRemoval shortcut to avoid hitting the JS/DOM-bridge for synchronous removal.

Reply:
yes

Reply:
I believe this can be marked as resolved after the above commit. 

Next level would be to to check that the parent isn't null and then use whatever parent to remove the node. However in that case, the dev has taken control of the DOM-element and moved it.